mw-server/internal/routers/favoriteUser.route_test.go

```diff
@@ -0,0 +1,54 @@
+package routers
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"mwserver/internal/auth"
+	"mwserver/internal/config"
+	"mwserver/internal/openapi"
+	"net/http"
+	"testing"
+
+	openapiGeneral "mwserver/apiAutogenerated/general"
+)
+
+func TestCreateFavoriteUser(t *testing.T) {
+	newConfig, err := config.LoadConfig("../../")
+	if err != nil {
+		t.Fatalf("Failed to load config: : %v", err)
+	}
+	generalApi := openapi.MakeGeneralAPIClient(&newConfig)
+	_, err = generalApi.DevAPI.DevResetDbGet(context.Background()).Execute()
+	if err != nil {
+		t.Fatalf("Failed to reset db: %v", err)
+	}
+
+	currentUserID := "7cdb041b-4574-4f7b-a500-c53e74c72e94"
+	donorUserUuid := currentUserID
+	acceptorUserUuid := currentUserID
+
+	t.Run("Should return 204 status and check that changes happened in DB", func(t *testing.T) {
+		token, err := auth.GenerateJWT(currentUserID, newConfig.SecretSessionKey)
+		if err != nil {
+			t.Fatalf("Failed to generate JWT: %v", err)
+		}
+		request := openapiGeneral.SchemasCreateFavoriteUserPayload{
+			AcceptorUserUuid: acceptorUserUuid,
+			DonorUserUuid:    donorUserUuid,
+		}
+		ctx := context.WithValue(context.Background(), auth.ContextKeyAuthorization, "Bearer "+token)
+		response, err := generalApi.FavoriteUserAPI.CreateFavoriteUser(ctx).Request(request).Execute()
+		if err != nil {
+			t.Fatalf("Failed to create favorite user: %v", err)
+		}
+		assert.Equal(t, http.StatusNoContent, response.StatusCode)
+
+		user, response, err := generalApi.UserAPI.GetUserByUuid(ctx, acceptorUserUuid).Execute()
+		if err != nil {
+			t.Fatalf("Failed to get user: %v", err)
+		}
+		favoriteUsers := user.FavoriteForUsers
+		assert.Contains(t, favoriteUsers, donorUserUuid, "Expected donorUserUuid to be in FavoriteForUsers")
```

Comment:
This code is not very readable because, instead of reading one line, you have to read two. It would be better to add user.FavoriteForUsers directly into assert.Contains.